app/controllers/publishers/promo_registrations_controller.rb

```diff
@@ -15,51 +49,99 @@ def overview
     start_date = (params[:month]&.to_date || Date.today).at_beginning_of_month
     end_date = start_date.at_end_of_month
 
-    aggregate_stats = PromoRegistration.stats_for_registrations(
-      promo_registrations: user.promo_registrations,
-      start_date: start_date,
-      end_date: end_date
-    )
+    groups = Rails.cache.fetch(GROUP_CACHE_KEY, expires_in: 1.hour) do
+      EyeshadeClient.referrals.groups
+    end
 
-    groups = Eyeshade::Referrals.new.groups
-    statement = Eyeshade::Referrals.new.statement(publisher: user, start_date: start_date, end_date: end_date)
+    group_counts = group_counts(groups, user, start_date, end_date)
 
     groups.each do |group|
-      group[:count] = statement.select { |s| s[:groupId] == group[:id] }.length
+      group[:counts] = group_counts.dig(group[:id]) || {}
+      group[:name] = I18n.t(".promo.dashboard.group", number: group[:name].remove("Group"))
     end
 
-    groups << october_2019_totals(start_date, aggregate_stats, statement)
+    groups << october_2019_totals(start_date, group_counts)
 
     render json: {
       groups: groups.compact,
-      totals: aggregate_stats,
+      totals: sum_counts(group_counts.values),
       lastUpdated: publisher_referrals_last_update(user),
     }
   end
 
   private
 
-  # In October 2019 we had an instance where users could receive confirmations with the legacy $5 rate along with the new group rates.
+  # Internal: Makes a request to Promo Server and aggregates the number of downloads or installs for a given user
+  #
+  # groups       - the group information from Eyeshade
+  # user         - the publisher to find their finalizations
+  # start_date   - the beginning of the range to recieve data within
+  # end_date     - the end of the range to recieve data within
+  #
+  # Returns a hash of group_ids and their associated counts, { group_id => count }
+  def group_counts(groups, user, start_date, end_date)
+    # Create a { country => group_id } mapping hash
+    countries = groups.each_with_object({}) do |group, hash|
+      group[:codes].each { |code| hash[code] = group[:id] }
```

Comment:
Is this part merged in yet to production?

Reply:
It's the section `PromoClient.reporting.geo_stats_by_referral_code`. It returns strictly countries right now but what we want is country codes. 🌐